baseapp/baseapp_reports/graphql/object_types.py

```diff
@@ -7,27 +7,56 @@
 
 from baseapp_core.graphql import DjangoObjectType, get_object_type_for_model
 
+from .filters import ReportTypeFilter
+
 Report = swapper.load_model("baseapp_reports", "Report")
+ReportType = swapper.load_model("baseapp_reports", "ReportType")
+
 
-ReportTypesEnum = graphene.Enum.from_enum(Report.ReportTypes)
+class ReportsCount(graphene.ObjectType):
+    counts = graphene.JSONString()
 
 
-def create_object_type_from_enum(name, enum):
-    fields = {}
-    for reaction_type in enum:
-        fields[reaction_type.name] = graphene.Int()
-    fields["total"] = graphene.Int()
-    return type(name, (graphene.ObjectType,), fields)
+class BaseReportTypeObjectType:
+    class Meta:
+        interfaces = (relay.Node,)
+        model = ReportType
+        fields = (
+            "id",
+            "name",
+            "label",
+            "content_types",
+            "sub_types",
+            "parent_type",
+        )
+        filterset_class = ReportTypeFilter
 
 
-ReportsCount = create_object_type_from_enum("ReportsCount", Report.ReportTypes)
+class ReportTypeObjectType(
+    BaseReportTypeObjectType, gql_optimizer.OptimizedDjangoObjectType, DjangoObjectType
+):
+    class Meta(BaseReportTypeObjectType.Meta):
+        pass
 
 
 class ReportsInterface(relay.Node):
     reports_count = graphene.Field(ReportsCount)
     reports = DjangoFilterConnectionField(get_object_type_for_model(Report))
     my_reports = graphene.Field(get_object_type_for_model(Report), required=False)
 
+    def resolve_reports_count(self, info, **kwargs):
```

Comment:
why do we need this resolve if we are saving it as a json field on the model itself? seems like we have a N+1 issue here that is easy to fix, seems like we could just delete this resolve method... I would also change
```reports_count = graphene.Field(ReportsCount)```
to be
```reports_count = GenericScalar()```
you can import `GenericScalar` with `from graphene.types.generic import GenericScalar`

this way on the frontend you can consume it with just `obj.reports_count.total` otherwise you would need to do something like `JSON.parse(obj.reports_count.count).total`

Reply:
Thank you for this suggestion, it really resolved this in a great way. I have modified to it